EricEReber/FFNN_optimization
# marriage

- add dummy figures
- Infer batch size in schedulers
- Remove old code
- Improve plotting
- Remove more dead code
- Simplify task_aa
- Rename task_aa
- add more comparisons for optimal parameter selection
- Add testing files
- Fix
- More testing
- Pog face multinomail implementation???
- More softmax more stable
- finish task a, fix momentum, add plot
- Fix a to z in backprop
- add test
- fix backpropagation bug
- Chad adam vs incel sigmoid
- split into class files
- refactor
- Fix resets
- for merge
- Momentum Ada
- Add mlptest
- for merge
- fix weight initialization
- Try to beat polynomial fit
- Add seed parameter
- Return scores dict
- for pulL
- Possible fix for schedulers
- Maybe fix batch size????
- More changes to task a to test
- Pretty big issue with batches maybe fixed
- add classification option
- Small fix
- Fix adam
- Add debug print
- Remove debug
- I am not crazy!
- We pray to Goodfellow et Al.
- add batch search
- Do item
- Structure for task B finished, task C in progress. The pipeline for task B is working fine, awaiting changes to task_a for further work. Plot terrain has also been added to utils.py
- task_c.py ready for testing, a final touch after 'optimize_scheduler' is finished
- task_b and task_c updated accordingly to the changes applied to optimize scheduler and optimize batch
- fix bugs add plot
- Add booststrap
- add docs
- Commite
- task_b and task_d updated to fit teh changes done in optimize scheduler
- copy-pasted FFNN from branch marriage to this one due to rebase confilct
- Clean up code, more comments
- bootstrap
- NO TEST_ACCS IN FFNNga task_d.py ga task_d.py
- Add ability to use the best weights
- fix optimize_batch
- Refactor scores dict from fit
- Added crossvalidation. Should work
- Fixing problems
- prep for cross_val
- i gotta go biology man
- i gotta go biology man
- plotting error or accuracy over model complexity implemented through method optimie_arch in utils.py
- Fix bugs related to crossvalidation
- Update crossval
- for merge
- cross validated
- fix file write
- Small print improvement
- Something Something
- Fix scaling of the data
- More on task d
- Remove testing dataset because of crossval
- Fix bug in crossval
- Work on mse by model complexity
- Remove bad import
- Add onehot encoding
- Sublcass Hessian added to schedulurs
- Functionality for Hessian added in line 159 in FFNN.py
- minor fix
- small fix
- Changes to Hessian
- reverted task_d
- Problems with Hessian continue
- Small fixes
- Remove Hessian from FFNN
- plots
- Add hessian code and move progress bar to utils
- Add verification file
- add time plot
- small change
- change write file
- adagrad supremecy
- Fixed hessian
- Fix bug in confusion matrix
- Hessian and plot over model complexity
- Make hessian crossvalidated
- Added wrong utils
- Grid search files and some figs
- important results
- Work more on task d and complexity
- new constant v hessian fig
- optimal params so far
- Fix title
- test momentum 1
- Some small changes
- Remove old code
- 66 66 66
- plot files
- Rename file
- Fix momentum shenanigans
- Update
- Update this to work
- fix NaN
- fix NaN
- Fit figs for regression fo Frankie
- This saved github from being destroyed by greg
- REALY LEAKY
- Titel change
- fontsize changes
- Working task_b.py
- Figs
- i hate this
- Update readme
- Add more comments to FFNN
- Just some figs
- Comments added to scripts
- Comment in Schedulers.py
- Comments in task_c.py
